tests/ProjectStorageBackendTest.php

```diff
@@ -90,6 +92,47 @@ public function testProjectStorageAssignBackend(string $backendName): void
         $this->client->purgeDeletedProject($project['id']);
     }
 
+    public function testProjectStorageAssignBackendFailedWithNonNumericBackendId(): void
+    {
+        $name = 'My org';
+        $organization = $this->client->createOrganization($this->testMaintainerId, [
+            'name' => $name,
+        ]);
+
+        $project = $this->client->createProject($organization['id'], [
+            'name' => 'My test',
+            'dataRetentionTimeInDays' => 1,
+        ]);
+
+        $apiPostMethod = new ReflectionMethod(ManageClient::class, 'apiPost');
+        $apiPostMethod->setAccessible(true);
+
+        try {
+            $apiPostMethod->invoke($this->client, '/manage/projects/' . $project['id'] . '/storage-backend', [
+                'storageBackendId' => 'non-numeric',
+            ]);
+            // @phpstan-ignore-next-line calling by `invoke` is not recognized by PHPStan as this is dynamic call by reflection
+        } catch (ClientException $e) {
+            $this->assertEquals(400, $e->getCode());
+            $this->assertEquals('storageBackendId must be an integer.', $e->getMessage());
+        }
+
+        try {
+            $apiPostMethod->invoke($this->client, '/manage/projects/' . $project['id'] . '/storage-backend', [
+                'storageBackendId' => '666',
+            ]);
+            // @phpstan-ignore-next-line calling by `invoke` is not recognized by PHPStan as this is dynamic call by reflection
+        } catch (ClientException $e) {
+            // backend storage '666' not found
+            // here we're only testing the format of `storageBackendId` so we don't care about the existence of backend storage
+            $this->assertEquals(500, $e->getCode());
```

Comment:
Pride mi, ze by to nemalo vratit Internal err, v pripade ze tam davas idcko ktore neexistuje. Druha vec je ze tento test https://github.com/keboola/kbc-manage-api-php-client/blob/20e8e1cd78bd40ceb3eb7830a34ab86233a6c04a/tests/StorageBackendTest.php#L176 bohuzial po sebe nemaze zaregistrovane backendy. Takze ked si vylistujes vsetky backendy https://keboolamanagementapi.docs.apiary.io/#reference/super-storage-backends-management/storage-backend-collection/list-backends napr na https://connection.e2e.us-east-1.aws.keboola.dev/ tak tam to Idcko teoreticky moze existovat. 

Takze test by si mal idealne zistit posledne idecko a testovat +1. Manage zatial nebezi paralelne takze by to asi slo. Pripadne ma napada, ze by to slo este otestovat tak, ze si zalozim backend rovno zmazem a to IDcko pouzijem na tento test.

Reply:
Vrací to `BackendNotFoundException`: https://github.com/keboola/connection/blob/3f73a3882fe203af129dc7f22dedfec230ceb738/legacy-app/application/src/Storage/Service/Backend/Assign/BackendAssign.php#L140

Kterou následně konvertuje `ExceptionConverter` na 500: https://github.com/keboola/connection/blob/3f73a3882fe203af129dc7f22dedfec230ceb738/legacy-app/application/src/Common/Exception/JsonExceptionConverter/ExceptionConverter.php#L65

Ale s tím ID+1 máš pravdu.